master/internal/taskconfigpolicy/task_config_policy.go

```diff
@@ -0,0 +1,41 @@
+package taskconfigpolicy
+
+import (
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+)
+
+// ResourceConstraints are non-overridable resource constraints.
+// Submitted workloads that request resource quanities exceeding defined resource constraints in a
+// given scope are rejected.
+type ResourceConstraints struct {
+	MaxSlots    *int `json:"max_slots"`
+	MaxPriority *int `json:"max_priority"`
```

Comment:
I think we want just `MaxSlots` here under `resources`. Then we can have `priority_limit`, `max_priority` and `min_priority` directly in the `Constraints` struct. `priority_limit` would come from the CLI or webUI and max/min would come from the database.

Reply:
Ooh gotcha that makes sense!
So the priority limit is the requested limit specified by the user, and is the max/min priority set in this go struct based on the RM type (so in the API handler)?

Reply:
Yeah, exactly! At least, that's how I envisioned it working ... we may change it as we get farther along. :)

Reply:
Cool! Does that leave room for the user to potentially try to fill in `max_priotiy` and `min_priority` if we use the Go `Constraints` struct to evaluate the integrity of their input config?

Reply:
Good point, it does! I hadn't thought of that. Thinking out loud ... we could define a separate priority struct?

```
type Constraints struct{
  ResourceConstraints
  RMPriority // working name; these are the min/max priorities in the database
  priority_limit // user input
}

type RMPriority struct{
  max_priority
  min_priority
```

or we could have a "user input" Constraints struct and a "database" Constraints struct. It just occurs to me that if we use this approach (two Constraint types) that we would have to translate between the two in the GET API. If we have an extra `RMPriority` struct, could we just drop that data in the GET before returning to the user?